### DISCUSSION TITLE
ATSPM server hardware requirement

### DISCUSSION THREAD

**Discussion by hzsp333** [udotdevelopment/ATSPM#217] [Q&A]
Hello,

We are looking for purchasing a database server for ATSPM. Is there any recommendations on the minimum / recommended requirements for the hardware of the server station (e.g. model, processor, memory, and if graphic card is needed, etc.)? Thanks!

**Comment by ChesireFreeThrow** [comment]
Other than the information in the [Installation manual](https://udottraffic.utah.gov/ATSPM/Images/ATSPM_Installation_Manual_4.3.pdf) it depends on how many controllers and how many logs you are going to have

**Reply by hzsp333** [reply]
Thanks for the answer. I searched through the manual as well as other sources but did not find much helpful information related to the hardware. At this stage, we are not going to cover a lot of intersections (probably 10 at most). Do you think a relatively cheaper server with no graphic card would be able to handle that? I am a bit concerned about generating charts. Not sure if we need two or more servers to handle all the tasks...

**Reply by avetraffic** [reply]
A cheap server with no graphics card should do fine for 10 signals, even with the charts. One server sounds like it should definitely be enough for your current needs.

**Reply by hzsp333** [reply]
Thank you so much for your information!

**Comment by avetraffic** [comment]
It also says this in the FAQ under System Requirements:

System Requirements: Operating Systems and Software: The UDOT Automated Signal Performance Measures system runs on Microsoft Windows Servers. The web components are hosted by Microsoft Internet Information Server(IIS). The database server is a Microsoft SQL 2016 server. Storage and Processing: Detector data uses about 40 % of the storage space of the UDOT system, so the number of detectors attached to a controller will have a huge impact on the amount of storage space required.Detector data is also the most important information we collect. We estimate that each signal will generate 19 MB of data per day. The amount of processing power required is highly dependant on how many signals are on the system, how many servers will be part of the system, and how many people will be using the system. It is possible to host all of the system functions on one powerful server, or split them out into multiple, less expensive servers. If your agency decided to make the Automated Signal Performance Measures available to the public, it might be best to have a web server separate from the database server.Much of the heavy processing for the charts is done by web services, and it is possible to host these services on a dedicated computer. While each agency should consult with their IT department for specific guidelines on how to best deliver a secure, stable and responsive solution, we can estimate that most mid-range to high-end servers will be able to handle the task of hosting and creating measures for most agencies.

Windows Server 2008 or newer installed
.NET 4.5.2 Framework installed
IIS 7 or better installed, along with ASP.NET 4.0 or later
SQL Server Express, SQL Server 2008 R2, or newer installed
Firewall exceptions for connections to the controllers
If Watchdog features are desired, installation requires access to an SMTP (email) server. It will accept email from the Automated Signal Performance Measures (ATSPM) server. The SMTP server can reside on the same machine.
Microsoft Visual Studio 2013 or later is recommended
